apps/web/src/store/use-env-controller.ts

```diff
@@ -0,0 +1,65 @@
+import { useQuery, useQueryClient } from 'react-query';
+import { IEnvironment } from '@novu/shared';
+import { getCurrentEnvironment, getMyEnvironments } from '../api/environment';
+import { useContext, useEffect, useState } from 'react';
+import { api } from '../api/api.client';
+import { AuthContext } from './authContext';
+import { QueryKeys } from '../api/query.keys';
+
+export type EnvironmentContext = {
+  readonly: boolean;
+  isLoading: boolean;
+  environment: IEnvironment | undefined;
+  setEnvironment: (environment: string) => void;
+};
+
+export const useEnvController = (): EnvironmentContext => {
+  const queryClient = useQueryClient();
+  const { setToken } = useContext(AuthContext);
+  const [readonly, setReadonly] = useState<boolean>(false);
```

Comment:
@ainouzgali @davidsoderberg  I've seen that the `useEnvController` was used multiple times rather than managing the provider context. Usually the controller is only important once just to abstract the context provider business logic. And later it shouldn't be imported directly since it creates more instances of the Context Data. We should use there `useContext(EnvironmentContext)` instead of using useEnvController

Reply:
It just might cause the issue where we have multiple instances of the "readonly" flag across the app. So keeping it in sync might cause some issues. Rather than only having a single copy of it in the provider context and just accessing it from there.